File: retaguarda-software-house/php/retaguarda-sh/src/model/cadastros/Empresa.php
<?php
/*******************************************************************************
Title: T2Ti ERP 3.0
Description: Model relacionado à tabela [EMPRESA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

class Empresa extends EloquentModel implements \JsonSerializable
{
    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'EMPRESA';

    /**
     * Gets e Sets
     */
    public function getIdAttribute() 
	{
		return $this->attributes['ID'];
	}

	public function setIdAttribute($id) 
	{
		$this->attributes['ID'] = $id;
	}

    public function getRazaoSocialAttribute() 
	{
		return $this->attributes['RAZAO_SOCIAL'];
	}

	public function setRazaoSocialAttribute($razaoSocial) 
	{
		$this->attributes['RAZAO_SOCIAL'] = $razaoSocial;
	}

    public function getNomeFantasiaAttribute() 
	{
		return $this->attributes['NOME_FANTASIA'];
	}

	public function setNomeFantasiaAttribute($nomeFantasia) 
	{
		$this->attributes['NOME_FANTASIA'] = $nomeFantasia;
	}

    public function getCnpjAttribute() 
	{
		return $this->attributes['CNPJ'];
	}

	public function setCnpjAttribute($cnpj) 
	{
		$this->attributes['CNPJ'] = $cnpj;
	}

    public function getInscricaoEstadualAttribute() 
	{
		return $this->attributes['INSCRICAO_ESTADUAL'];
	}

	public function setInscricaoEstadualAttribute($inscricaoEstadual) 
	{
		$this->attributes['INSCRICAO_ESTADUAL'] = $inscricaoEstadual;
	}

    public function getInscricaoMunicipalAttribute() 
	{
		return $this->attributes['INSCRICAO_MUNICIPAL'];
	}

	public function setInscricaoMunicipalAttribute($inscricaoMunicipal) 
	{
		$this->attributes['INSCRICAO_MUNICIPAL'] = $inscricaoMunicipal;
	}

    public function getTipoRegimeAttribute() 
	{
		return $this->attributes['TIPO_REGIME'];
	}

	public function setTipoRegimeAttribute($tipoRegime) 
	{
		$this->attributes['TIPO_REGIME'] = $tipoRegime;
	}

    public function getCrtAttribute() 
	{
		return $this->attributes['CRT'];
	}

	public function setCrtAttribute($crt) 
	{
		$this->attributes['CRT'] = $crt;
	}

    public function getDataConstituicaoAttribute() 
	{
		return $this->attributes['DATA_CONSTITUICAO'];
	}

	public function setDataConstituicaoAttribute($dataConstituicao) 
	{
		$this->attributes['DATA_CONSTITUICAO'] = $dataConstituicao;
	}

    public function getTipoAttribute() 
	{
		return $this->attributes['TIPO'];
	}

	public function setTipoAttribute($tipo) 
	{
		$this->attributes['TIPO'] = $tipo;
	}

    public function getEmailAttribute() 
	{
		return $this->attributes['EMAIL'];
	}

	public function setEmailAttribute($email) 
	{
		$this->attributes['EMAIL'] = $email;
	}

    public function getAliquotaPisAttribute() 
	{
		return $this->attributes['ALIQUOTA_PIS'];
	}

	public function setAliquotaPisAttribute($aliquotaPis) 
	{
		$this->attributes['ALIQUOTA_PIS'] = $aliquotaPis;
	}

    public function getAliquotaCofinsAttribute() 
	{
		return $this->attributes['ALIQUOTA_COFINS'];
	}

	public function setAliquotaCofinsAttribute($aliquotaCofins) 
	{
		$this->attributes['ALIQUOTA_COFINS'] = $aliquotaCofins;
	}

    public function getLogradouroAttribute() 
	{
		return $this->attributes['LOGRADOURO'];
	}

	public function setLogradouroAttribute($logradouro) 
	{
		$this->attributes['LOGRADOURO'] = $logradouro;
	}

    public function getNumeroAttribute() 
	{
		return $this->attributes['NUMERO'];
	}

	public function setNumeroAttribute($numero) 
	{
		$this->attributes['NUMERO'] = $numero;
	}

    public function getComplementoAttribute() 
	{
		return $this->attributes['COMPLEMENTO'];
	}

	public function setComplementoAttribute($complemento) 
	{
		$this->attributes['COMPLEMENTO'] = $complemento;
	}

    public function getCepAttribute() 
	{
		return $this->attributes['CEP'];
	}

	public function setCepAttribute($cep) 
	{
		$this->attributes['CEP'] = $cep;
	}

    public function getBairroAttribute() 
	{
		return $this->attributes['BAIRRO'];
	}

	public function setBairroAttribute($bairro) 
	{
		$this->attributes['BAIRRO'] = $bairro;
	}

    public function getCidadeAttribute() 
	{
		return $this->attributes['CIDADE'];
	}

	public function setCidadeAttribute($cidade) 
	{
		$this->attributes['CIDADE'] = $cidade;
	}

    public function getUfAttribute() 
	{
		return $this->attributes['UF'];
	}

	public function setUfAttribute($uf) 
	{
		$this->attributes['UF'] = $uf;
	}

    public function getFoneAttribute() 
	{
		return $this->attributes['FONE'];
	}

	public function setFoneAttribute($fone) 
	{
		$this->attributes['FONE'] = $fone;
	}

    public function getContatoAttribute() 
	{
		return $this->attributes['CONTATO'];
	}

	public function setContatoAttribute($contato) 
	{
		$this->attributes['CONTATO'] = $contato;
	}

    public function getCodigoIbgeCidadeAttribute() 
	{
		return $this->attributes['CODIGO_IBGE_CIDADE'];
	}

	public function setCodigoIbgeCidadeAttribute($codigoIbgeCidade) 
	{
		$this->attributes['CODIGO_IBGE_CIDADE'] = $codigoIbgeCidade;
	}

    public function getCodigoIbgeUfAttribute() 
	{
		return $this->attributes['CODIGO_IBGE_UF'];
	}

	public function setCodigoIbgeUfAttribute($codigoIbgeUf) 
	{
		$this->attributes['CODIGO_IBGE_UF'] = $codigoIbgeUf;
	}

    public function getLogotipoAttribute() 
	{
		return $this->attributes['LOGOTIPO'];
	}

	public function setLogotipoAttribute($logotipo) 
	{
		$this->attributes['LOGOTIPO'] = $logotipo;
	}

    public function getRegistradoAttribute() 
	{
		return $this->attributes['REGISTRADO'];
	}

	public function setRegistradoAttribute($registrado) 
	{
		$this->attributes['REGISTRADO'] = $registrado;
	}

    public function getNaturezaJuridicaAttribute() 
	{
		return $this->attributes['NATUREZA_JURIDICA'];
	}

	public function setNaturezaJuridicaAttribute($naturezaJuridica) 
	{
		$this->attributes['NATUREZA_JURIDICA'] = $naturezaJuridica;
	}

    public function getSimeiAttribute() 
	{
		return $this->attributes['SIMEI'];
	}

	public function setSimeiAttribute($simei) 
	{
		$this->attributes['SIMEI'] = $simei;
	}

    public function getEmailPagamentoAttribute() 
	{
		return $this->attributes['EMAIL_PAGAMENTO'];
	}

	public function setEmailPagamentoAttribute($emailPagamento) 
	{
		$this->attributes['EMAIL_PAGAMENTO'] = $emailPagamento;
	}

    public function getDataRegistroAttribute() 
	{
		return $this->attributes['DATA_REGISTRO'];
	}

	public function setDataRegistroAttribute($dataRegistro) 
	{
		$this->attributes['DATA_REGISTRO'] = $dataRegistro;
	}

    public function getHoraRegistroAttribute() 
	{
		return $this->attributes['HORA_REGISTRO'];
	}

	public function setHoraRegistroAttribute($horaRegistro) 
	{
		$this->attributes['HORA_REGISTRO'] = $horaRegistro;
	}


    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
            isset($objeto->id) ? $this->setIdAttribute($objeto->id) : $this->setIdAttribute(null);

			$this->setRazaoSocialAttribute($objeto->razaoSocial);
			$this->setNomeFantasiaAttribute($objeto->nomeFantasia);
			$this->setCnpjAttribute($objeto->cnpj);
			$this->setInscricaoEstadualAttribute($objeto->inscricaoEstadual);
			$this->setInscricaoMunicipalAttribute($objeto->inscricaoMunicipal);
			$this->setTipoRegimeAttribute($objeto->tipoRegime);
			$this->setCrtAttribute($objeto->crt);
			$this->setDataConstituicaoAttribute($objeto->dataConstituicao);
			$this->setTipoAttribute($objeto->tipo);
			$this->setEmailAttribute($objeto->email);
			$this->setAliquotaPisAttribute($objeto->aliquotaPis);
			$this->setAliquotaCofinsAttribute($objeto->aliquotaCofins);
			$this->setLogradouroAttribute($objeto->logradouro);
			$this->setNumeroAttribute($objeto->numero);
			$this->setComplementoAttribute($objeto->complemento);
			$this->setCepAttribute($objeto->cep);
			$this->setBairroAttribute($objeto->bairro);
			$this->setCidadeAttribute($objeto->cidade);
			$this->setUfAttribute($objeto->uf);
			$this->setFoneAttribute($objeto->fone);
			$this->setContatoAttribute($objeto->contato);
			$this->setCodigoIbgeCidadeAttribute($objeto->codigoIbgeCidade);
			$this->setCodigoIbgeUfAttribute($objeto->codigoIbgeUf);
			$this->setLogotipoAttribute($objeto->logotipo);
			$this->setRegistradoAttribute($objeto->registrado);
			$this->setNaturezaJuridicaAttribute($objeto->naturezaJuridica);
			$this->setSimeiAttribute($objeto->simei);
			$this->setEmailPagamentoAttribute($objeto->emailPagamento);
			$this->setDataRegistroAttribute($objeto->dataRegistro);
			$this->setHoraRegistroAttribute($objeto->horaRegistro);

			// vincular objetos
        }
    }

    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getIdAttribute(),
			'razaoSocial' => $this->getRazaoSocialAttribute(),
			'nomeFantasia' => $this->getNomeFantasiaAttribute(),
			'cnpj' => $this->getCnpjAttribute(),
			'inscricaoEstadual' => $this->getInscricaoEstadualAttribute(),
			'inscricaoMunicipal' => $this->getInscricaoMunicipalAttribute(),
			'tipoRegime' => $this->getTipoRegimeAttribute(),
			'crt' => $this->getCrtAttribute(),
			'dataConstituicao' => $this->getDataConstituicaoAttribute(),
			'tipo' => $this->getTipoAttribute(),
			'email' => $this->getEmailAttribute(),
			'aliquotaPis' => $this->getAliquotaPisAttribute(),
			'aliquotaCofins' => $this->getAliquotaCofinsAttribute(),
			'logradouro' => $this->getLogradouroAttribute(),
			'numero' => $this->getNumeroAttribute(),
			'complemento' => $this->getComplementoAttribute(),
			'cep' => $this->getCepAttribute(),
			'bairro' => $this->getBairroAttribute(),
			'cidade' => $this->getCidadeAttribute(),
			'uf' => $this->getUfAttribute(),
			'fone' => $this->getFoneAttribute(),
			'contato' => $this->getContatoAttribute(),
			'codigoIbgeCidade' => $this->getCodigoIbgeCidadeAttribute(),
			'codigoIbgeUf' => $this->getCodigoIbgeUfAttribute(),
			'logotipo' => $this->getLogotipoAttribute(),
			'registrado' => $this->getRegistradoAttribute(),
			'naturezaJuridica' => $this->getNaturezaJuridicaAttribute(),
			'simei' => $this->getSimeiAttribute(),
			'emailPagamento' => $this->getEmailPagamentoAttribute(),
        ];
    }
}

File: retaguarda-software-house/delphi/retaguarda_sh/Model/Cadastros/Empresa.pas
{*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [EMPRESA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************}
unit Empresa;

interface

uses
  MVCFramework.Serializer.Commons, ModelBase;

type

  [MVCNameCase(ncLowerCase)]
  TEmpresa = class(TModelBase)
  private
    FId: Integer;
    FRazaoSocial: string;
    FNomeFantasia: string;
    FCnpj: string;
    FInscricaoEstadual: string;
    FInscricaoMunicipal: string;
    FTipoRegime: string;
    FCrt: string;
    FDataConstituicao: TDateTime;
    FTipo: string;
    FEmail: string;
    FEmailPagamento: string;
    FLogradouro: string;
    FNumero: string;
    FComplemento: string;
    FCep: string;
    FBairro: string;
    FCidade: string;
    FUf: string;
    FFone: string;
    FContato: string;
    FCodigoIbgeCidade: Integer;
    FCodigoIbgeUf: Integer;
    FLogotipo: string;
    FRegistrado: string;
    FNaturezaJuridica: string;
    FSimei: string;
    FDataRegistro: TDateTime;
    FHoraRegistro: string;

  public
//    procedure ValidarInsercao; override;
//    procedure ValidarAlteracao; override;
//    procedure ValidarExclusao; override;

    [MVCColumnAttribute('ID', True)]
		[MVCNameAsAttribute('id')]
		property Id: Integer read FId write FId;
    [MVCColumnAttribute('RAZAO_SOCIAL')]
		[MVCNameAsAttribute('razaoSocial')]
		property RazaoSocial: string read FRazaoSocial write FRazaoSocial;
    [MVCColumnAttribute('NOME_FANTASIA')]
		[MVCNameAsAttribute('nomeFantasia')]
		property NomeFantasia: string read FNomeFantasia write FNomeFantasia;
    [MVCColumnAttribute('CNPJ')]
		[MVCNameAsAttribute('cnpj')]
		property Cnpj: string read FCnpj write FCnpj;
    [MVCColumnAttribute('INSCRICAO_ESTADUAL')]
		[MVCNameAsAttribute('inscricaoEstadual')]
		property InscricaoEstadual: string read FInscricaoEstadual write FInscricaoEstadual;
    [MVCColumnAttribute('INSCRICAO_MUNICIPAL')]
		[MVCNameAsAttribute('inscricaoMunicipal')]
		property InscricaoMunicipal: string read FInscricaoMunicipal write FInscricaoMunicipal;
    [MVCColumnAttribute('TIPO_REGIME')]
		[MVCNameAsAttribute('tipoRegime')]
		property TipoRegime: string read FTipoRegime write FTipoRegime;
    [MVCColumnAttribute('CRT')]
		[MVCNameAsAttribute('crt')]
		property Crt: string read FCrt write FCrt;
    [MVCColumnAttribute('DATA_CONSTITUICAO')]
		[MVCNameAsAttribute('dataConstituicao')]
		property DataConstituicao: TDateTime read FDataConstituicao write FDataConstituicao;
    [MVCColumnAttribute('TIPO')]
		[MVCNameAsAttribute('tipo')]
		property Tipo: string read FTipo write FTipo;
    [MVCColumnAttribute('EMAIL')]
		[MVCNameAsAttribute('email')]
		property Email: string read FEmail write FEmail;
    [MVCColumnAttribute('EMAIL_PAGAMENTO')]
		[MVCNameAsAttribute('emailPagamento')]
		property EmailPagamento: string read FEmailPagamento write FEmailPagamento;
    [MVCColumnAttribute('LOGRADOURO')]
		[MVCNameAsAttribute('logradouro')]
		property Logradouro: string read FLogradouro write FLogradouro;
    [MVCColumnAttribute('NUMERO')]
		[MVCNameAsAttribute('numero')]
		property Numero: string read FNumero write FNumero;
    [MVCColumnAttribute('COMPLEMENTO')]
		[MVCNameAsAttribute('complemento')]
		property Complemento: string read FComplemento write FComplemento;
    [MVCColumnAttribute('CEP')]
		[MVCNameAsAttribute('cep')]
		property Cep: string read FCep write FCep;
    [MVCColumnAttribute('BAIRRO')]
		[MVCNameAsAttribute('bairro')]
		property Bairro: string read FBairro write FBairro;
    [MVCColumnAttribute('CIDADE')]
		[MVCNameAsAttribute('cidade')]
		property Cidade: string read FCidade write FCidade;
    [MVCColumnAttribute('UF')]
		[MVCNameAsAttribute('uf')]
		property Uf: string read FUf write FUf;
    [MVCColumnAttribute('FONE')]
		[MVCNameAsAttribute('fone')]
		property Fone: string read FFone write FFone;
    [MVCColumnAttribute('CONTATO')]
		[MVCNameAsAttribute('contato')]
		property Contato: string read FContato write FContato;
    [MVCColumnAttribute('CODIGO_IBGE_CIDADE')]
		[MVCNameAsAttribute('codigoIbgeCidade')]
		property CodigoIbgeCidade: Integer read FCodigoIbgeCidade write FCodigoIbgeCidade;
    [MVCColumnAttribute('CODIGO_IBGE_UF')]
		[MVCNameAsAttribute('codigoIbgeUf')]
		property CodigoIbgeUf: Integer read FCodigoIbgeUf write FCodigoIbgeUf;
    [MVCColumnAttribute('LOGOTIPO')]
		[MVCNameAsAttribute('logotipo')]
		property Logotipo: string read FLogotipo write FLogotipo;
    [MVCColumnAttribute('REGISTRADO')]
		[MVCNameAsAttribute('registrado')]
		property Registrado: string read FRegistrado write FRegistrado;
    [MVCColumnAttribute('NATUREZA_JURIDICA')]
		[MVCNameAsAttribute('naturezaJuridica')]
		property NaturezaJuridica: string read FNaturezaJuridica write FNaturezaJuridica;
    [MVCColumnAttribute('SIMEI')]
		[MVCNameAsAttribute('simei')]
		property Simei: string read FSimei write FSimei;
    [MVCColumnAttribute('DATA_REGISTRO')]
		[MVCNameAsAttribute('dataRegistro')]
		property DataRegistro: TDateTime read FDataRegistro write FDataRegistro;
    [MVCColumnAttribute('HORA_REGISTRO')]
		[MVCNameAsAttribute('horaRegistro')]
		property HoraRegistro: string read FHoraRegistro write FHoraRegistro;

  end;

implementation

{ TEmpresa }



end.